### PR TITLE
avoid NPE in createWaypointMarker (fix #8696)

### LINKED ISSUE #8696: NPE in MapMarkerUtils
Reported on GooglePlay as new problem in 2020.07.26-RC. Currently 2 users on Android 9 and 10.

```
java.lang.NullPointerException: 

at cgeo.geocaching.utils.MapMarkerUtils.createWaypointMarker (MapMarkerUtils.java:151)

at cgeo.geocaching.utils.MapMarkerUtils.getWaypointMarker (MapMarkerUtils.java:131)

at cgeo.geocaching.maps.mapsforge.v6.caches.AbstractCachesOverlay.getWaypointItem (AbstractCachesOverlay.java:385)

at cgeo.geocaching.maps.mapsforge.v6.caches.AbstractCachesOverlay.addItem (AbstractCachesOverlay.java:197)

at cgeo.geocaching.maps.mapsforge.v6.caches.SinglePointOverlay.fill (SinglePointOverlay.java:44)

at cgeo.geocaching.maps.mapsforge.v6.caches.SinglePointOverlay.lambda$IsbKcIoziki_SC79-DtI8GsJki0 (Unknown Source)

at cgeo.geocaching.maps.mapsforge.v6.caches.-$$Lambda$SinglePointOverlay$IsbKcIoziki_SC79-DtI8GsJki0.run (Unknown Source:2)

at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call (ScheduledDirectTask.java:41)

at io.reactivex.rxjava3.internal.schedulers.ScheduledDirectTask.call (ScheduledDirectTask.java:28)

at java.util.concurrent.FutureTask.run (FutureTask.java:266)

at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:301)

at java.util.concurrent.ThreadPoolExecutor.processTask (ThreadPoolExecutor.java:1187)

at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1152)

at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)

at java.lang.Thread.run (Thread.java:929)
```

ˋˋˋ

### PR DESCRIPTION
**Pull request by moving-bits**

### CODE REVIEW

**Comment by bekuno** [comment]
retest this please

**Comment by moving-bits** [comment]
retest this please

File: main/src/cgeo/geocaching/maps/MapOptions.java [hunk 1]
```diff
@@ -38,6 +38,9 @@ public MapOptions(final Context context, @Nullable final Bundle extras) {
             waypointType = (WaypointType) extras.get(Intents.EXTRA_WPTTYPE);
             mapState = extras.getParcelable(Intents.EXTRA_MAPSTATE);
             title = extras.getString(Intents.EXTRA_TITLE);
+            if (null != coords && null == waypointType) {
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/maps/MapOptions.java:41]: Under what conditions might this happen?

Comment (moving-bits) [reply main/src/cgeo/geocaching/maps/MapOptions.java:41]: This is the $1,000 question (and the reason why I called this a mitigation).
When I debug backwards from the error position I can see that this is the condition which leads to the error. I only do not know where it origins from. Also I cannot reproduce it. So I decided to go for a mitigation for now.

Comment (Lineflyer) [reply main/src/cgeo/geocaching/maps/MapOptions.java:41]: maybe imported waypoints from a GPX?

Comment (Lineflyer) [reply main/src/cgeo/geocaching/maps/MapOptions.java:41]: But you have to consider, that this is a new crash with current beta, which makes my previous comment unlikely.

Comment (moving-bits) [reply main/src/cgeo/geocaching/maps/MapOptions.java:41]: May be. Do you happen to have some test data for this case?

Comment (Lineflyer) [reply main/src/cgeo/geocaching/maps/MapOptions.java:41]: No, it was just a guess.

**Comment by Lineflyer** [comment]
Lets use the mitigation.

**Pull request merged**